### PR TITLE
[test] Add unit tests for ConditionalBuilder failure modes

### PR DESCRIPTION
**Pull request by croyzor**

### CODE REVIEW

File: src/builder/conditional.rs [hunk 1]
```diff
@@ -266,4 +266,28 @@ mod test {
...
+    fn test_not_all_cases() -> Result<(), BuildError> {
+        let predicate_inputs = vec![type_row![]; 2];
+        let mut builder = ConditionalBuilder::new(predicate_inputs, type_row![], type_row![])?;
+        n_identity(builder.case_builder(0)?)?;
+        match builder.finish_sub_container() {
+            Err(BuildError::ConditionalError(ConditionalBuildError::NotAllCasesBuilt {
+                ..
+            })) => Ok(()),
+            _ => panic!(),
+        }
```
Comment (ss2165) [inline src/builder/conditional.rs:280]: you can use `assert_matches!` from `cool_asserts`

**Review by ss2165 — COMMENTED** [review]
looks good, can just avoid some boiler plate

File: src/builder/conditional.rs (hunk 1 above)
Comment (croyzor) [reply src/builder/conditional.rs:280]: I tried, but `assert_matches` requires that the `BuildHandle` implements Debug.
I could just call `map(|_| ())` on the success case though...

**Review by ss2165 — APPROVED** [review]
👍

**Pull request merged**